### DISCUSSION TITLE
Suggestion: Provide a "Like" feature

### DISCUSSION THREAD

**Discussion by gyger** [posativ/isso#786] [Ideas]
On medium it is common to like or heart a page, also in the past I have seen webpages that have small hearts next to their articles, just to show that you appreciated the content. It is simpler than a comment and still shows that you enjoyed the text.
It would be nice if something like this would be possible with ISSO. I don't want to load another javascript library to enable something like this.

Perhaps it is not a goal of the project, than just close the report, otherwise it would be interesting if a function like this is welcome.

**Comment by csleary** [comment]
The up/down votes serve this purpose though, don't they?

**Comment by gyger** [comment]
Not to upvote a comment, but to like a post.
e.g. on this post on medium: https://backchannel.com/its-2016-where-s-my-flying-pizza-683ff360bedc?source=reading_list---------1-3
You can like it at the end of the post by pressing the heart, on medium this also shares the article to your follower, this is something Isso would not implement, but only count how many people pressed the heart.

**Comment by csleary** [comment]
Ah I see, sorry. You might be able to repurpose the comment counter to reflect a separate post/like count.

**Comment by ix5** [comment]
This would be quite neat. For other inspiration, I'd suggest looking at [svbtle](https://svbtle.com/)'s *kudos* button - in action e.g. at [dcurt.is](https://dcurt.is/).

**Comment by ix5** [comment]
@gyger https://github.com/ix5/isso/tree/thanks-mvp
![output](https://user-images.githubusercontent.com/10212877/105162738-4c725680-5b13-11eb-8c83-30a1f395da30.gif)

**Comment by gyger** [comment]
@ix5 Yes this is exactly what I was thinking of. Something to say thank you, without having to write a lot.

**Comment by ix5** [comment]
Feel free to take the code and make a workable PR out of it ;)

You'll need to work out a few kinks and make a few decisions that I haven't made yet:
- Proper endpoints (`reactions`? `thanks`? `likes`?) and consistent naming
- Tests...
- Decide on a database schema that makes more sense than mine (do you want only one type of reaction or extensibility?)
- How configurable should the client side be? Do you want to be able to define a custom icon? ...
- Styling of button so that it fits in with the rest of Isso
- Admin integration to view counts

**Comment by gyger** [comment]
Thanks, I am currently a little bit busy, but will keep it in the task list. If someone else wants to make a PR out of it feel free.

**Comment by ioistired** [comment]
I'd also appreciate a few more reactions like Disqus does:
![screenshot of reaction buttons, upvote, funny, love, surprised, angry, and sad, with a count of each reaction](https://user-images.githubusercontent.com/538336/106782698-18695c00-6610-11eb-8eeb-7ad411734825.png)

**Comment by ix5** [comment]
See also: https://github.com/posativ/isso/issues/582

Unless someone steps up, it will be unlikely this ever finds its way into Isso.

**Comment by ix5** [comment]
Moving to "Ideas" since no one wants to work on this.
